src/crypto/public_key/rsa.js

```diff
@@ -274,5 +313,169 @@ export default {
+    const pBNum = new BN(p);
+    const qBNum = new BN(q);
+    const dBNum = new BN(d);
+    const dq = dBNum.mod(qBNum.subn(1)); // d mod (q-1)
+    const dp = dBNum.mod(pBNum.subn(1)); // d mod (p-1)
+    const sign = nodeCrypto.createSign(enums.read(enums.hash, hash_algo));
+    sign.write(data);
+    sign.end();
+    if (typeof nodeCrypto.createPrivateKey !== 'undefined') { //from version 11.6.0 Node supports der encoded key objects
+      const der = RSAPrivateKey.encode({
+        version: 0,
+        modulus: new BN(n),
+        publicExponent: new BN(e),
+        privateExponent: new BN(d),
+        // switch p and q
+        prime1: new BN(q),
+        prime2: new BN(p),
+        exponent1: dq,
+        exponent2: dp,
+        coefficient: new BN(u)
+      }, 'der');
+      return new Uint8Array(sign.sign({ key: der, format: 'der', type: 'pkcs1' }));
+    }
+    const pem = RSAPrivateKey.encode({
+      version: 0,
+      modulus: new BN(n),
+      publicExponent: new BN(e),
+      privateExponent: new BN(d),
+      // switch p and q
+      prime1: new BN(q),
+      prime2: new BN(p),
+      exponent1: dq,
+      exponent2: dp,
+      coefficient: new BN(u)
+    }, 'pem', {
+      label: 'RSA PRIVATE KEY'
+    });
+    return new Uint8Array(sign.sign(pem));
+  },
+
+  bnVerify: async function (hash_algo, s, n, e, hashed) {
+    n = new BN(n);
+    s = new BN(s);
+    e = new BN(e);
+    if (n.cmp(s) <= 0) {
+      throw new Error('Signature size cannot exceed modulus size');
+    }
+    const nred = new BN.red(n);
+    const EM1 = s.toRed(nred).redPow(e).toArrayLike(Uint8Array, 'be', n.byteLength());
+    const EM2 = await pkcs1.emsa.encode(hash_algo, hashed, n.byteLength());
+    return util.Uint8Array_to_hex(EM1) === EM2;
+  },
+
+  webCryptoVerify: async function (hash_name, data, s, n, e) {
+    const jwk = publicToJwk(n, e);
+    const key = await webCrypto.importKey("jwk", jwk, {
+      name: "RSASSA-PKCS1-v1_5",
+      hash: { name:  hash_name }
+    }, false, ["verify"]);
+    return webCrypto.verify({ "name": "RSASSA-PKCS1-v1_5" }, key, s, data);
+  },
+
+  nodeCryptoVerify: async function (hash_algo, data, s, n, e) {
+    const verify = nodeCrypto.createVerify(enums.read(enums.hash, hash_algo));
+    verify.write(data);
+    verify.end();
+    if (typeof nodeCrypto.createPrivateKey !== 'undefined') { //from version 11.6.0 Node supports der encoded key objects
+      const der = RSAPublicKey.encode({
+        modulus: new BN(n),
+        publicExponent: new BN(e)
+      }, 'der');
+      try {
+        return verify.verify({ key: der, format: 'der', type: 'pkcs1' }, s);
+      } catch (err) {
+        return false;
+      }
+    }
+    const key = RSAPublicKey.encode({
+      modulus: new BN(n),
+      publicExponent: new BN(e)
+    }, 'pem', {
+      label: 'RSA PUBLIC KEY'
+    });
+    try {
+      return await verify.verify(key, s);
+    } catch (err) {
+      return false;
+    }
```

Comment:
And this as well. Maybe deduplicate both the object and the try/catch, if possible.